### PR TITLE
Upgrade Microsoft.Identity.Client package reference

### PR DESCRIPTION
**Pull request by philon-msft**

### CODE REVIEW

File: src/CacheIdentityClient.cs [hunk 1]
```diff
@@ -27,12 +28,12 @@ internal class CacheIdentityClient : ICacheIdentityClient
     private readonly Func<bool, Task<AuthenticationResult>> _getToken;
 
     internal static ICacheIdentityClient CreateForSystemAssignedManagedIdentity()
-        => new CacheIdentityClient(ManagedIdentityApplicationBuilder.Create()
+        => new CacheIdentityClient(ManagedIdentityApplicationBuilder.Create(ManagedIdentityId.SystemAssigned)
             .WithExperimentalFeatures()
```
Comment (neha-bhargava) [inline src/CacheIdentityClient.cs:32]: `.WithExperimentalFeatures()` is no longer needed.

**Review by neha-bhargava — APPROVED** [review]

File: src/CacheIdentityClient.cs [hunk 2]
```diff
@@ -27,12 +28,12 @@ internal class CacheIdentityClient : ICacheIdentityClient
...
             .Build());
 
     internal static ICacheIdentityClient CreateForUserAssignedManagedIdentity(string clientId)
-        => new CacheIdentityClient(ManagedIdentityApplicationBuilder.Create(clientId)
+        => new CacheIdentityClient(ManagedIdentityApplicationBuilder.Create(ManagedIdentityId.WithUserAssignedClientId(clientId))
```
[the remark this answers is not in the document]
Comment (gladjohn) [reply src/CacheIdentityClient.cs:35]: if you can't parse the string to a GUID then it's a resource id in string format

**Comment by gladjohn** [comment]
changes looks good. please enable [MSAL logging](https://learn.microsoft.com/en-us/azure/active-directory/develop/msal-logging-dotnet), this will help us with troubleshooting any issues you may see.

File: src/CacheIdentityClient.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (gladjohn) [reply src/CacheIdentityClient.cs:35]: Or if you want the users to be specific then, better to ask them what they are providing. But that would mean an API change at your end. something you can plan for vnext

**Comment by philon-msft** [comment]
> changes looks good. please enable [MSAL logging](https://learn.microsoft.com/en-us/azure/active-directory/develop/msal-logging-dotnet), this will help us with troubleshooting any issues you may see.

Thanks @gladjohn - we plan to enable MSAL logging in an upcoming release once StackExchange.Redis itself supports a logging story

**Review by gladjohn — APPROVED** [review]

File: src/AzureCacheForRedis.cs [hunk 3]
```diff
@@ -41,7 +41,7 @@ public static async Task<ConfigurationOptions> ConfigureForAzureWithSystemAssign
     /// Throws on failure by default (configurable in the <see cref="ConfigureForAzureAsync"/> method).
     /// </summary>
     /// <param name="configurationOptions">The configuration to update.</param>
-    /// <param name="clientId">Client ID of the user-assigned managed identity.</param>
+    /// <param name="clientId">Client ID or resource ID of the user-assigned managed identity.</param>
     /// <param name="principalId">Principal (object) ID of the user-assigned managed identity.</param>
```
Comment (gladjohn) [inline src/AzureCacheForRedis.cs:45]: MSAL does not support principalId for Managed Identity. Is this something you want us to support?

Comment (philon-msft) [reply src/AzureCacheForRedis.cs:45]: @samsaha-ms can you comment on this? We pass the MI's object/principal ID as the username for the Redis connection, so I believe we use it in some capacity while validating tokens on the server side

Comment (samsaha-ms) [reply src/AzureCacheForRedis.cs:45]: Yes, this is something related to what our extension does to interact with Redis, not related to MSAL.
It does two things 1. Acquire token from MSAL 2. Send AUTH request to Redis with principal-id and token to comply with redis command format "auth <username> <password>"and we validate this pair in server side.
We could actually drop this field from here, we could parse the acquired token and get the oid field in token and use that value while sending to redis.

**Review by samsaha-ms — APPROVED** [review]

**Pull request merged**